services:
  postgres:
    image: postgres:17.2-alpine3.21
    container_name: postgres-test
    restart: always
    environment:
      POSTGRES_DB: postgres
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: test
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U myuser"]
      interval: 30s
      timeout: 10s
      retries: 5
    networks:
      - dodgygeezers_test

  registration:
      # restart: "no"
    ports:
      - 8000:8000
      - 5678:5678
      - 50051:50051
    build:
      context: .
      dockerfile: Dockerfile
    container_name: registration-microservice-test
    volumes:
    - ./app:/app/app
    - ./pyproject.toml:/app/pyproject.toml
    - ./uv.lock:/app/uv.lock
    # command: sleep infinity  # Infinite loop to keep container alive doing nothing
    command:
      - python3
      - -Xfrozen_modules=off
      - -m
      - debugpy
      - --listen
      - 0.0.0.0:5678
      - --wait-for-client
      - -m
      - uvicorn
      - app.main:app
      - --reload
      - --host
      - "0.0.0.0"
      - --port
      - "8000"
    env_file:
      - .env.development
    environment:
      - DB_HOST=postgres
      - PYDEVD_DISABLE_FILE_VALIDATION=1
    depends_on:
      - postgres
    networks:
      - dodgygeezers_test
  
  adminer:
    image: adminer:4.8.1-standalone
    container_name: adminer-test
    ports:
      - "8080:8080"
    networks:
      - dodgygeezers_test

volumes:
  postgres_data:

networks:
  dodgygeezers_test:
    # driver: bridge
    external: true
